src/pages/Earn/useFarmRegistry.ts

```diff
@@ -89,15 +107,60 @@ export const useFarmRegistry = () => {
           tvlUSD: farmData[e.returnValues.stakingAddress].tvlUSD,
           rewardsUSDPerYear: farmData[e.returnValues.stakingAddress].rewardsUSDPerYear,
           isFeatured: !!featuredPoolWhitelist[e.returnValues.stakingAddress],
+          rewardApr: undefined,
+          swapApr: undefined,
+          apr: undefined,
+          apy: undefined,
         })
       })
 
     farmSummaries
       .sort((a, b) => Number(fromWei(toBN(b.rewardsUSDPerYear).sub(toBN(a.rewardsUSDPerYear)))))
       .sort((a, b) => Number(fromWei(toBN(b.tvlUSD).sub(toBN(a.tvlUSD)))))
 
-    setFarmSummaries(farmSummaries)
-  }, [kit.web3.eth])
+    const results = await Promise.all(
+      farmSummaries.map((summary) => {
+        return client.query({ query: pairDataGql, variables: { id: summary.lpAddress.toLowerCase() } })
+      })
+    )
+    const farmInfos = results.map((result: ApolloQueryResult<any>, index) => {
+      let swapRewardsUSDPerYear = 0
+      const summary = farmSummaries[index]
+      const { loading, error, data } = result
+      if (!loading && !error && data) {
+        const lastDayVolumeUsd = data.pair.pairHourData.reduce(
+          (acc: number, curr: { hourlyVolumeUSD: string }) => acc + Number(curr.hourlyVolumeUSD),
+          0
+        )
+        swapRewardsUSDPerYear = Math.floor(lastDayVolumeUsd * 365 * 0.0025)
+      }
+      const rewardApr = new Percent(summary.rewardsUSDPerYear, summary.tvlUSD)
+      const swapApr = new Percent(toWei(swapRewardsUSDPerYear.toString()), summary.tvlUSD)
+      const apr = new Percent(
+        toBN(toWei(swapRewardsUSDPerYear.toString())).add(toBN(summary.rewardsUSDPerYear)).toString(),
```

Comment:
No need to do conversions when both of these are numbers.

```
new Percent(swapRewardsUSDPerYear + summary.rewardsUSDPerYear).toString()
```

Reply:
summary.rewardsUSDPerYear is not number and it's string
When we convert this to Number, we sometimes get values like 2.4252818037259246e+23
And then we face such issue like this
![image](https://user-images.githubusercontent.com/95660534/166849817-78311128-6039-4f31-9782-966678841bd8.png)

Reply:
`swapRewardsUSDPerYear` should not be in wei units if you are typing as a `Number`.

Use regular units for `Number`
and decimal'd units (like 18 decimals, wei, 9 decimals) in bignumbers